gutenTAG/base_oscillations/custom_input.py

```diff
@@ -88,6 +89,9 @@ def generate_only_base(self,
 
         if len(df) < length:
             raise ValueError("Number of rows in the input timeseries file is less than the desired length")
+        if df.dtypes[0] == 'int64':
+            df = df.astype(float)
+            warnings.warn("Input data was of integer type and has been automatically converted to float.")
```

Comment:
Can we make this more general?

```suggestion
        col_type = df.dtypes[0]
        if col_type != np.float_:
            df = df.astype(float)
            warnings.warn(f"Input data was of {col_type} type and has been automatically converted to float.")
```

I know that this might fail if the input type cannot be cast to float, but it would fail later on anyway. If we force the type here, the user will get the error message early on.